### PR TITLE
Dashboard interactive views

### LINKED ISSUE #1687: Introduce interactive views in the dashboard
During the last few month, we've spent a lot of time improving the general experience of Saleor's dashboard. A huge progress was made, but we've also started to think that Django's static views are no longer capable of providing the UX that we'd like to see in Saleor. We could try to use more AJAX views and load partial HTML templates to create dynamic and interactive pieces of UI, but in the long run, this would lead to a codebase which is difficult to maintain and doesn't meet the modern standards.

After internal discussions, a decision was made to start introducing client-rendered dynamic views in the dashboard. Since this is a significant change, we'd like to do it on a separate branch and start with adopting a single view as a proof-of-concept. The view we could start with is the category list -
 `/dashboard/categories/` (this is a general issue for discussion, I'd open a separate one where we'll gather requirements for rewriting this particular view).

I'd like to agree on the technological stack to be used. Basically, Saleor will serve as a backend accessed through the GraphQL API. The client-side toolset would be:

- React
- Redux
- React Router (v4?)
- Apollo Client - as a GraphQL API
- [Material-UI](https://github.com/mui-org/material-ui)
- [redux-form](https://redux-form.com/7.2.1/) + [redux-form-material-ui](https://github.com/erikras/redux-form-material-ui)

If there's anything else that we already know we should use, write it in the comments and I'll update this list once we agree. Any other thoughts and feedback on this issue are welcomed.

### PR DESCRIPTION
**Pull request by dominik-zeglen**
Resolves #1687 

TODO:
- [x] Hide filter bar
- [x] Translations
- [ ] Zero page
- [ ] Query caching
- [ ] Code cleanup
- [ ] Sorting / filtering API 
- [ ] Get fully recursed category tree

### CODE REVIEW

File: saleor/graphql/product/types.py [hunk 1]
```diff
@@ -245,4 +245,5 @@ def resolve_attributes(category_pk):
         queryset = queryset.filter(
             Q(product_types__in=product_types) |
             Q(product_variant_types__in=product_types))
+    print(queryset.distinct())
```
Comment (rafalp) [inline saleor/graphql/product/types.py]: Remains of debugging session? ;)

File: saleor/static/dashboard/js/components/app/category/details/subcategoryList.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  <Card style={styles.cardSubcategories}>
+    <CardContent>
+      {props.category && (
+        <div>
+          <CardTitle>Subcategories</CardTitle>
+          <FlatButton color={'secondary'} style={{ margin: '2rem 0 1rem' }}>Dodaj</FlatButton>
+        </div>
+      )}
+      <Table style={props.category ? styles.table.childCategory : styles.table.rootCategory}>
+        <TableHead adjustForCheckbox={false}
```
Comment (rafalp) [inline saleor/static/dashboard/js/components/app/category/details/subcategoryList.js]: I'm wondering if we shouldn't find other way to format multiline tags in our JSX, seeing how this falls under the same "annoying to realign on rename ascii-art" category we've went trough the html templates and our python codebase.

Comment (patrys) [reply saleor/static/dashboard/js/components/app/category/details/subcategoryList.js]: I believe it's pretty common for JSX to use:

```typescript
<Foo
  bar="1"
  baz="2"
/>
```

Comment (rafalp) [reply saleor/static/dashboard/js/components/app/category/details/subcategoryList.js]: Yup, that's what [AirB&B style guide recommends](https://github.com/airbnb/javascript/tree/master/react#tags), and it seems to be only popular guide around AFAIR.

File: saleor/graphql/product/types.py (hunk 1 above)
Comment (dominik-zeglen) [reply saleor/graphql/product/types.py]: Definitely feature ;)

**Comment by dominik-zeglen** [comment]
I need API for:
- creating, updating and deleting category
- listing all categories (my implementation isn't as good as we want it to be)

**Comment by patrys** [comment]
Do we need all of these components to use `withRouter`?

**Comment by dominik-zeglen** [comment]
I'll be refactoring this soon, but I'm pretty we can pass `history` prop to all components using any form of routing. Current code is glue and power tape at this moment.

**Comment by NyanKiyoshi** [comment]
I feel like the category creation form should redirect us to the parent category of the newly created category instead of the new category. Just like we are currently doing on dashboard 1.0.

I think, that most of the time, we are creating a bunch of categories for a given base category, then, a bunch of categories for the subcategories.

I will be waiting on your thoughts on this.

**Pull request merged**